sisl/geometry.py

```diff
@@ -1019,27 +1021,451 @@ def overlap(self, other, eps=0.1, offset=(0., 0., 0.), offset_other=(0., 0., 0.)
+           sort coordinates according to lattice vectors, if a tuple of
+           ints is passed it will be equivalent to ``sort(lattice0=lattice[0], lattice1=lattice[1])``.
+           This is different behaviour than `numpy.lexsort`!
+           Note that before sorting we multiply the fractional coordinates by the length of the
+           lattice vector. This ensures that `atol` is meaningful for both `axis` and `lattice` since
+           they will be on the same order of magnitude.
+        vector : (3, ), optional
+           sort along a user defined vector, similar to `lattice` but with a user defined
+           direction. Note that `lattice` sorting and `vector` sorting are *only* equivalent
+           when the lattice vector is orthogonal to the other lattice vectors.
+        group : {'Z', 'symbol', 'tag', 'species'} or (str, ...), optional
+           group together a set of atoms by various means.
+           `group` may be one of the listed strings.
+           For ``'Z'`` atoms will be grouped in atomic number
+           For ``'symbol'`` atoms will be grouped by their atomic symbol.
+           For ``'tag'`` atoms will be grouped by their atomic tag.
+           For ``'species'`` atoms will be sorted according to their specie index.
+           If a tuple/list is passed the first item is described. All subsequent items are a
+           list of groups, where each group comprises elements that should be sorted on an
+           equal footing. If one of the groups is None, that group will be replaced with all
+           non-mentioned elements. See examples.
+        func : callable, optional
+           pass a sorting function which should have an interface like ``func(geometry, atom, **kwargs)``.
+           The first argument is the geometry to sort. The 2nd argument is a list of indices in
+           the current group of sorted atoms. And ``**kwargs`` are any optional arguments
+           currently collected, i.e. `ascend`, `atol` etc.
+           The function should return either a list of atoms, or a list of list of atoms (in which
+           case the atomic indices have been split into several groups that will be sorted individually
+           for subsequent sorting methods).
+           In either case the returned indices must never hold any other indices but the ones passed
+           as ``atom``.
+        ascend, descend : bool, optional
+            control ascending or descending sorting for all subsequent sorting methods.
+            Default ``ascend=True``.
+        atol : float, optional
+            absolute tolerance when sorting numerical arrays for subsequent sorting methods.
+            When a selection of sorted coordinates are grouped via `atol`, we ensure such
+            a group does not alter its indices. I.e. the group is *always* ascending indices.
+            Note this may have unwanted side-effects if `atol` is very large.
+            Default ``1e-9``.
```

Comment:
What is `very large`? Compared to what?

Reply:
I have added ` is very large compared
            to the difference between atomic coordinates.`